classes/ObjectModel.php

```diff
@@ -1460,12 +1485,18 @@ public function addFieldsRequiredDatabase($fields)
             return false;
         }
 
-        if (!Db::getInstance()->execute('DELETE FROM '._DB_PREFIX_.'required_field WHERE object_name = \''.get_class($this).'\'')) {
+        $objectName = $this->getObjectName();
+        if (!Db::getInstance()->execute(
+            'DELETE FROM ' . _DB_PREFIX_ . "required_field WHERE object_name = '$objectName'")
```

Comment:
`$objectName` MUST be escaped. I think you can use `DB::getInstance()->escape()` for that.